EIPS/eip-7867.md

```diff
@@ -0,0 +1,565 @@
+    * Wallets MUST reject `rollback` when used in a `none` batch, even if the
+      batch is upgraded to `loose` or `strict` atomicity. This also applies to
+      calls that do not specify an explicit `onFailure` mode.
+* MAY reject (with error code `ROLLBACK_EXPECTED`) the request if the batch is
+  expected to be rolled back.
+* SHOULD inform the user before executing any calls if any call in the batch is
+  expected to fail.
+
+
+#### `wallet_getCallsStatus`
+
+When `wallet_getCallsStatus` is called with a batch identifier corresponding to
+a batch submitted with the batch-scope `flowControl` capability enabled, the
+following changes override the behaviour defined in EIP-5792. Note that:
+
+* There are no changes when called with a batch without this capability enabled;
+  and
+* Even if the behaviour of the batch is not changed from the default (eg.
+  setting `atomicity` to `strict` and omitting the `flowControl` capability for
+  all calls), the following changes still apply.
+
+##### Removed Requirements
+
+These requirements defined in EIP-5792 are removed:
+
+> * If a wallet executes multiple calls **atomically** in a single transaction,
+>   `wallet_getCallsStatus` MUST return an object with a `receipts` field that
+>   contains a single transaction receipt, corresponding to the transaction
+>   in which the calls were included.
+> * If a wallet executes multiple calls **non-atomically** through some
+>   `capability` defined elsewhere, `wallet_getCallsStatus` MUST return an
+>   object with a `receipts` field that contains **an array of receipts** for
+>   all transactions containing batch calls that were included onchain. This
+>   includes the batch calls that were included on-chain but eventually
+>   reverted.
+
+
+##### Added Requirements
+
+###### Capabilities
+
+The returned capabilities object:
+
+* MUST contain a `flowControl` key set to exactly the boolean `true`.
+    * It may be tempting to include additional detail about the status of
+      individual calls here, but don't. Instead use a multi-status capability
+      defined elsewhere.
+
+###### Receipts
+
+The returned `receipts` array:
+
+* MUST NOT contain more than one receipt for the same transaction.
+* SHOULD NOT contain receipts for transactions without a call from the requested
+  batch.
+* MUST contain exactly one receipt capturing each successful call.
+    * Multiple calls MAY be captured in one receipt, but the successful
+      execution of one call MUST NOT be captured by multiple receipts.
+    * Given two calls (_A_ and _B_) in a batch, the following are non-exhaustive
+      example combinations of calls-per-receipt. Each `(...)` is a receipt from
+      a single transaction.
+        * Valid Examples:
+            * `[(successful A, successful B)]`
+            * `[(successful A), (successful B)]`
+            * `[(successful A, unsuccessful B), (successful B)]`
+            * `[(unsuccessful A), (successful A), (successful B)]`
```

Comment:
how are unsuccessful calls represented? obviously the receipt as a whole has a status, but you can have failed calls in a successful receipt (requiring tracing to track down what actually happened) - and I don't see where in the [call result](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-5792.md?plain=1#L164) it's identified one way or the other